lisa/tools/lspci.py

```diff
@@ -184,6 +209,19 @@ def get_gpu_devices(self, force_run: bool = False) -> List[PciDevice]:
         ]
         return gpu_device_list
 
+    def get_devices_by_vendor_device_id(
+        self,
+        vendor_id: str,
+        device_id: str,
+        force_run: bool = False,
+    ) -> List[PciDevice]:
+        full_list = self.get_devices(force_run)
+        devices_list = []
+        for device in full_list:
+            if device.device_id == device_id and device.vendor_id == vendor_id:
```

Comment:
lspci tool is implemented in a way where it will cache the list of all devices. I dont want to run multiple lspci combinations with different args. This way, we can enhance the PciDevice class properties as well.